Automattic/woocommerce-payments
# Do not redirect to the onboarding page when account retrieval fails.

Fixes #2511

We've had a number of reports of unexpected redirects happening when navigating to WCPay admin pages:
> When clicking on "deposits", "transactions", or "disputes" it does not go to the report. Instead, the user gets a finish setup page. When they click the "finish setup" button it goes to the set up dashboard even if they have it configured. Customer can't see the deposit, transaction, or dispute history.

This looks related to an [previous PR](https://github.com/Automattic/woocommerce-payments/pull/2275) I raised which was released in `2.7.0`.

**I have not been able to reproduce this behaviour locally** but it [was noted here](https://wordpress.org/support/topic/keeps-asking-to-finish-setup/#post-14690904) that rolling back to a version earlier than `2.7.0` fixes the issue.

Please also be aware of https://github.com/Automattic/woocommerce-payments/pull/2514 which fixes a related issue #2513

#### Changes proposed in this Pull Request

Pass `true` to `is_stripe_connected()` to prevent a redirect when account retrieval fails. This contrasts to the current behaviour on `develop` which will redirect to `/payments/connect` when account retrieval fails.

#### Testing instructions

Add the following code to a mu-plugin:

```php
add_filter( 'option_wcpay_account_data', function ( $wcpay_account_data ) {
	if ( 0 === wp_rand( 0, 1 ) ) {
		return $wcpay_account_data;
	}

	return [
		'expires' => time() + DAY_IN_SECONDS,
		'account' => 'ERROR',
	];
} );
```

This introduces a 50% chance to simulate account retrieval failure.

Then:

- Navigate to `/wp-admin/admin.php?page=wc-admin&path=/payments/overview`
- Observe 50% of the time you will observe "Error determining the connection status." in the Account details box. The other 50% of the time you will see the expected account details. Refresh the page a few times to experience both scenarios.

Compare this with the behaviour on `develop`. i.e. switch back to `develop` with the above mu-plugin still in place.

You'll notice that 50% of the time you're redirected to `/payments/connect` and the other 50% of the time you will see the expected account details.

-------------------
